features/include/pcl/features/impl/integral_image2D.hpp

```diff
@@ -189,7 +189,7 @@ IntegralImage2D<DataType, Dimension>::computeIntegralImages (
   {
     SecondOrderType* so_previous_row = &second_order_integral_image_[0];
     SecondOrderType* so_current_row  = so_previous_row + (width_ + 1);
-    memset (so_previous_row, 0, sizeof (SecondOrderType) * (width_ + 1));
+    *so_previous_row = ElementType::Zero(width_ + 1);
```

Comment:
I'm sorry for my unfamiliarity with Eigen library. I made these errors with `pcl/features/include/pcl/features/impl/integral_image2D.hpp:192:22` when building. 

What else I should do to solve the errors? Thank you!
```text
/usr/include/eigen3/Eigen/src/Core/AssignEvaluator.h: In instantiation of 'void Eigen::internal::call_assignment_no_alias(Dst&, const Src&, const Func&) [with Dst = Eigen::Matrix<double, 6, 1, 0, 6, 1>; Src = Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Matrix<double, 3, 1> >; Func = Eigen::internal::assign_op<double, double>]':
/usr/include/eigen3/Eigen/src/Core/AssignEvaluator.h:804:27:   required from 'void Eigen::internal::call_assignment(Dst&, const Src&, const Func&, typename Eigen::internal::enable_if<(! Eigen::internal::evaluator_assume_aliasing<Src>::value), void*>::type) [with Dst = Eigen::Matrix<double, 6, 1, 0, 6, 1>; Src = Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Matrix<double, 3, 1> >; Func = Eigen::internal::assign_op<double, double>; typename Eigen::internal::enable_if<(! Eigen::internal::evaluator_assume_aliasing<Src>::value), void*>::type = void*]'
/usr/include/eigen3/Eigen/src/Core/AssignEvaluator.h:782:18:   required from 'void Eigen::internal::call_assignment(Dst&, const Src&) [with Dst = Eigen::Matrix<double, 6, 1, 0, 6, 1>; Src = Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Matrix<double, 3, 1> >]'
/usr/include/eigen3/Eigen/src/Core/PlainObjectBase.h:714:32:   required from 'Derived& Eigen::PlainObjectBase<Derived>::_set(const Eigen::DenseBase<OtherDerived>&) [with OtherDerived = Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Matrix<double, 3, 1> >; Derived = Eigen::Matrix<double, 6, 1, 0, 6, 1>]'
/usr/include/eigen3/Eigen/src/Core/Matrix.h:225:24:   required from 'Eigen::Matrix<_Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols>& Eigen::Matrix<_Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols>::operator=(const Eigen::DenseBase<OtherDerived>&) [with OtherDerived = Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Matrix<double, 3, 1> >; _Scalar = double; int _Rows = 6; int _Cols = 1; int _Options = 0; int _MaxRows = 6; int _MaxCols = 1]'
pcl/features/include/pcl/features/impl/integral_image2D.hpp:192:22:   required from 'void pcl::IntegralImage2D<DataType, Dimension>::computeIntegralImages(const DataType*, unsigned int, unsigned int) [with DataType = float; unsigned int Dimension = 3]'
pcl/features/include/pcl/features/impl/integral_image2D.hpp:67:3:   required from 'void pcl::IntegralImage2D<DataType, Dimension>::setInput(const DataType*, unsigned int, unsigned int, unsigned int, unsigned int) [with DataType = float; unsigned int Dimension = 3]'
pcl/features/include/pcl/features/impl/integral_image_normal.hpp:133:23:   required from 'void pcl::IntegralImageNormalEstimation<PointInT, PointOutT>::initCovarianceMatrixMethod() [with PointInT = pcl::PointXYZ; PointOutT = pcl::Normal]'
pcl/features/src/integral_image_normal.cpp:46:3:   required from here
/usr/include/eigen3/Eigen/src/Core/AssignEvaluator.h:833:3: error: static assertion failed: YOU_MIXED_MATRICES_OF_DIFFERENT_SIZES
  833 |   EIGEN_STATIC_ASSERT_SAME_MATRIX_SIZE(ActualDstTypeCleaned,Src)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
so_previous_row is of type `SecondOrderType`. And you're assigning it to `ElementType`. It's just a matrix size mismatch

Reply:
I fixed it. Thanks! :satisfied: I'll pay more attention to the code context next time.